scripts/silo/decorate_models_by_relation.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env sentry exec
+
+import django.apps
+from django.db.models.fields.related_descriptors import ForwardManyToOneDescriptor
+
+from scripts.silo.common import apply_decorators
+from sentry.models import Organization
+
+"""
+This is an alternative to add_mode_limits.py that uses an algorithmic definition of
+the silos and aims for 100% coverage. It examines the fields of model classes and
+uses a graph traversal algorithm to find all models that point to the `Organization`
+model, either directly or through a number of steps. Those models are tagged for the
+customer silo, and all others for the control silo.
+
+Instructions for use:
+
+1. Commit or stash any Git changes in progress.
+2. From the Sentry project root, do
+     ./scripts/silo/decorate_models_by_relation.py
+3. Do `git status` or `git diff` to observe the results. Commit if you're happy.
+"""
+
+
+def get_sentry_model_classes():
+    for model_class in django.apps.apps.get_models():
+        if model_class._meta.app_label == "sentry":
+            yield model_class
+
+
+def get_related_models(model_class):
+    for attr_name in dir(model_class):
+        attr = getattr(model_class, attr_name)
+        if isinstance(attr, ForwardManyToOneDescriptor):
```

Comment:
Does this cover many to many associations as well?

Reply:
Right, I wanted to go back and play around with the models some more. I'll check on that.

Reply:
Done: bb30d9d55985905c04543887285af67d14a46926

Turns out that was pretty important. It changes the output for about a dozen models. Thanks for the reminder.